### PR TITLE
Add withTimeout to Chat and Offline Messages 

### LINKED ISSUE #1224: API calls where messages are sent to offline stores exceeding intended timeouts
@placer14 [recently committed some code](https://github.com/OpenBazaar/openbazaar-go/commit/37a31e2e1b8654f55f087bfe7ca217bd3cfec026) to reduce the timeout period, for mobile servers, where the node will attempt to send an online message to a peer to 5 seconds. If the timeout is hit, an offline message will be sent via a background goroutine, and any API should return an immediate `200` response while the message is sent in the background.

I was testing this out by trying to send a `POST /ob/follow` API call to `QmTmCkNLUcPGvf3mSYDme4UQudgn9oCVqE13GHnrF6sjLj` (who I confirmed was offline). The call took `64833 ms` to return a response. This seems to indicate that the timeout isn't working as intended in mobile and perhaps desktop (where the timeout should be 30 seconds).

### PR DESCRIPTION
**Pull request by hoffmabc**
Helps fix #1224

### CODE REVIEW

**Comment by hoffmabc** [comment]
cc: @drwasho 

@placer14 @cpacia please take a closer look at this as I added some more timeouts to offline messaging. The reason for this is that mobile is also having trouble with offline timeouts related to purchases. I've tested this locally on desktop and it seems to be performing as intended, but I'm not sure if you guys will find side effects this may impact.

Below are two purchase calls that are hitting the 5s limit (modified the start for desktop to use 5s) and falling through.

image

image

File: core/net.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (n *OpenBazaarNode) SendOfflineMessage(p peer.ID, k *libp2p.PubKey, m *pb.M
 	log.Debugf("Sending offline message to: %s, Message Type: %s, PointerID: %s, Location: %s", p.Pretty(), m.MessageType.String(), pointer.Cid.String(), pointer.Value.Addrs[0].String())
 	OfflineMessageWaitGroup.Add(2)
 	go func() {
-		ctx, cancel := context.WithCancel(context.Background())
+		ctx, cancel := context.WithTimeout(context.Background(), n.OfflineMessageFailoverTimeout)
```
Comment (cpacia) [inline core/net.go]: Since this is in a separate goroutine I don't think we need a timeout here.

Comment (hoffmabc) [reply core/net.go]: Are we good on the other ones?

Comment (cpacia) [reply core/net.go]: I think the other ones are OK. If you remove this one we can merge.

**Comment by hoffmabc** [comment]
GTG @cpacia

**Pull request merged**